h2o-core/src/main/java/water/rapids/ast/prims/filters/dropduplicates/DropDuplicateRows.java

```diff
@@ -48,7 +47,7 @@ public Frame dropDuplicates() {
               .outputFrame(Key.make(), sourceFrame.names(), sourceFrame.domains())); // Removing duplicates, domains remain the same
 
       outputFrame = Merge.sort(deDuplicatedFrame, deDuplicatedFrame.numCols() - 1);
-      outputFrame.remove(outputFrame.numCols() - 1);
+      Scope.track(outputFrame.remove(outputFrame.numCols() - 1));
```

Comment:
Obvious leak. The vec gets removed from the frame and scope only tracks the frame. Our scopes do not add all the keys inside once Scope.track is invoked, they only add the top Keyed element and then, cascade is called just in time of deletion.